### PR TITLE
feat: update btc/eth to use historical tx history and new ws payloads

### PR DESCRIPTION
**Pull request by kaladinlight**

### CODE REVIEW

File: packages/chain-adapters/src/ChainAdapterCLI.ts [hunk 1]
```diff
@@ -28,216 +24,229 @@ const unchainedUrls = {
     wsUrl: 'wss://dev-api.bitcoin.shapeshift.com'
   },
   [ChainTypes.Ethereum]: {
-    httpUrl: 'https://dev-api.ethereum.shapeshift.com',
-    wsUrl: 'wss://dev-api.ethereum.shapeshift.com'
+    httpUrl: 'http://api.ethereum.localhost',
```
Comment (kaladinlight) [inline packages/chain-adapters/src/ChainAdapterCLI.ts:27]: revert to dev endpoints

File: packages/chain-adapters/src/ChainAdapterCLI.ts [hunk 2]
```diff
@@ -28,216 +24,229 @@ const unchainedUrls = {
...
+  }
+}
+
+const main = async () => {
+  try {
+    const chainAdapterManager = new ChainAdapterManager(unchainedUrls)
+
+    const wallet = await getWallet()
+
+    //await testBitcoin(chainAdapterManager, wallet)
```
Comment (kaladinlight) [inline packages/chain-adapters/src/ChainAdapterCLI.ts:247]: uncomment tests

File: packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts [hunk 3]
```diff
@@ -360,22 +402,17 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Ethereum> {
     onMessage: (msg: chainAdapters.Transaction<ChainTypes.Ethereum>) => void,
     onError: (err: chainAdapters.SubscribeError) => void
   ): Promise<void> {
-    const { wallet, bip44Params = ChainAdapter.defaultBIP44Params } = input
+    const { bip44Params = ChainAdapter.defaultBIP44Params } = input
 
-    const address = await this.getAddress({ wallet, bip44Params })
+    //const address = await this.getAddress({ wallet, bip44Params })
```
Comment (kaladinlight) [inline packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts:407]: use derived address

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts [hunk 4]
```diff
@@ -90,10 +83,73 @@ export class ChainAdapter
...
+
+    const txs = await Promise.all(
+      (data.txs ?? []).map(async (tx) => {
+        const addresses = getAddresses(tx).filter((addr) => accountAddresses.includes(addr))
+
+        return await Promise.all(
+          addresses.map(async (addr) => {
+            const parsedTx = await this.parser.parse(tx, addr)
+
+            return {
```
Comment (elmutt) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:123]: could this be shortened by spreading parsedTx?

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts [hunk 5]
```diff
@@ -377,7 +427,13 @@ export class ChainAdapter
           fee: tx.fee,
           status: getStatus(tx.status),
           tradeDetails: tx.trade,
-          transfers,
+          transfers: tx.transfers.map((transfer) => ({
+            assetId: transfer.assetId,
```
Comment (elmutt) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:431]: could shorten with a spread?

File: packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts [hunk 6]
```diff
@@ -115,14 +123,48 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Ethereum> {
...
+      pageSize: input.pageSize,
+      cursor: input.cursor
+    })
+
+    const txs = await Promise.all(
+      data.txs.map(async (tx) => {
+        const parsedTx = await this.parser.parse(tx, input.pubkey)
+
+        return {
+          address: input.pubkey,
```
Comment (elmutt) [inline packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts:140]: shorten with spread?

**Review by cjthompson — DISMISSED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**